jax/ops/scatter.py

```diff
@@ -334,4 +341,19 @@ def segment_sum(data, segment_ids, num_segments=None,
 
   out = jnp.zeros((num_segments,) + data.shape[1:], dtype=data.dtype)
   segment_ids = jnp.mod(segment_ids, num_segments)
-  return index_add(out, segment_ids, data, indices_are_sorted, unique_indices)
+
+  num_buckets = 1 if bucket_size is None \
+                  else -(-segment_ids.size // bucket_size)  # ceil_division
+  if num_buckets == 1:
```

Comment:
In general I am in two minds about this kind of fast path. In the absence of `jit` it certainly makes sense to have the fast path. In the presence of `jit` it just makes the code more complicated!

No action required, just something to keep in mind.

Reply:
Make sense to me. I think a default that gives correctness at reasonable performance cost is better than `None`. But we can do that in a separate PR.